### DISCUSSION TITLE
Starlette OAuth causes invalid auth argument on HTTPX 0.18.2

### DISCUSSION THREAD

**Discussion by alessandroralc** [encode/httpx#1712] [General]
After upgrading to HTTPX 0.18.2, the starlette oauth module started to crash.

When i call the oauth authorize_redirect method, i get the following error: Invalid \"auth\" argument: <httpx._config.UnsetType object at 0x7fb5ac920310>"}

In HTTPX 0.18.1, everything works like a charm.

Could you verify? Thanks in advance.

**Comment by lovelydinosaur** [comment]
One helpful thing to do here would be to point us at the specific project/repository - I'm not sure I'd know offhand what "the starlette oauth module" is exactly, without a bit of digging around.

It *looks?* like that package is using some private implementation detail that it shouldn't be doing, and so has broken on the newer `httpx` version.

Note that `httpx._config.UnsetType` includes an underscore on the `_config` portion, which implies private API space which shouldn't be accessed directly.

I'd expect that the package will want to be using `httpx.USE_CLIENT_DEFAULT`, which now exists as a piece of public API.

**Reply by alessandroralc** [reply]
The Starlette Oauth Module is provided by the AuthLib -> https://pypi.org/project/Authlib/
I'm using the version 0.15.4 (most recent)

**Reply by alessandroralc** [reply]
@tomchristie , i discovered that already exists an issue on Authlib github project that is solving this problem ( https://github.com/lepture/authlib/commit/40276708374fcc7c0ee6b1ce4955dbe7db82f5ba ). Thanks for the attention.
